pkg/blocks/fetchuri.go

```diff
@@ -0,0 +1,309 @@
+	}
+
+	// If there is no cleanup step or if this step is the cleanup step, exit.
+	if tmpl.CleanupStep.IsZero() || f.Type == StepCleanup {
+		return nil
+	}
+
+	// Create a CleanupStep instance and add it to the FetchURIStep instance.
+	logging.L().Debugw("step", "name", tmpl.Name)
+	cleanup, err := f.MakeCleanupStep(&tmpl.CleanupStep)
+	logging.L().Debugw("step", zap.Error(err))
+	if err != nil {
+		logging.L().Errorw("error creating cleanup step", zap.Error(err))
+		return err
+	}
+
+	f.CleanupStep = cleanup
+
+	return nil
+}
+
+// GetType returns the type of the step as StepType.
+func (f *FetchURIStep) GetType() StepType {
+	return StepFetchURI
+}
+
+// Cleanup is a method to establish a link with the Cleanup interface.
+// Assumes that the type is the cleanup step and is invoked by
+// f.CleanupStep.Cleanup.
+func (f *FetchURIStep) Cleanup(execCtx TTPExecutionContext) (*ActResult, error) {
+	result, err := f.Execute(execCtx)
+	if err != nil {
+		return nil, err
+	}
+	return &result.ActResult, err
+}
+
+// GetCleanup returns a slice of CleanupAct if the CleanupStep is not nil.
+func (f *FetchURIStep) GetCleanup() []CleanupAct {
+	if f.CleanupStep != nil {
+		return []CleanupAct{f.CleanupStep}
+	}
+	return []CleanupAct{}
+}
+
+// ExplainInvalid returns an error message explaining why the FetchURIStep
+// is invalid.
+//
+// **Returns:**
+//
+// error: An error message explaining why the FetchURIStep is invalid.
+func (f *FetchURIStep) ExplainInvalid() error {
+	var err error
+	if f.FetchURI == "" {
+		err = errors.New("empty FetchURI provided")
+	}
+
+	if f.Location == "" && err != nil {
+		err = errors.New("empty Location provided")
+	}
+
+	if f.Name != "" && err != nil {
+		err = fmt.Errorf("[!] invalid FetchURIStep: [%s] %v", f.Name, zap.Error(err))
+	}
+
+	return err
+}
+
+// IsNil checks if the FetchURIStep is nil or empty and returns a boolean value.
+func (f *FetchURIStep) IsNil() bool {
+	switch {
+	case f.Act.IsNil():
+		return true
+	case f.FetchURI == "":
+		return true
+	case f.Location == "":
+		return true
+	default:
+		return false
+	}
+}
+
+// Execute runs the FetchURIStep and returns an error if any occur.
+func (f *FetchURIStep) Execute(execCtx TTPExecutionContext) (*ExecutionResult, error) {
+	logging.L().Info("========= Executing ==========")
+
+	if err := f.fetchURI(execCtx); err != nil {
+		logging.L().Error(zap.Error(err))
+		return nil, err
+	}
+
+	logging.L().Info("========= Result ==========")
+
+	return &ExecutionResult{}, nil
+}
+
+// fetchURI executes the FetchURIStep with the specified Location, Uri, and additional arguments,
+// and an error if any errors occur.
+func (f *FetchURIStep) fetchURI(execCtx TTPExecutionContext) error {
+	appFs := f.FileSystem
+	absLocal := f.Location
+
+	if appFs == nil {
+		var err error
+		appFs = afero.NewOsFs()
+		absLocal, err = FetchAbs(f.Location, f.WorkDir)
+		if err != nil {
+			return err
+		}
+	}
+
+	if ok, _ := afero.Exists(appFs, absLocal); ok && !f.Overwrite {
```

Comment:
let's not squash the error here:

```suggestion
	exists, err := afero.Exists(appFs, absLocal)
	if err != nil {
	  return nil, err 
	}
	if ok && !f.Overwrite { 
```

Reply:
Under the hood the error is just from Stat

```
// Check if a file or directory exists.
func Exists(fs Fs, path string) (bool, error) {
	_, err := fs.Stat(path)
	if err == nil {
		return true, nil
	}
	if os.IsNotExist(err) {
		return false, nil
	}
	return false, err
}
```
Just a path error, which in this case we ignore since this is to check for existence. The next error chance you get is when you try to create the file.